cmd/experimental/kjobctl/pkg/cmd/create/create_test.go

```diff
@@ -1131,6 +1131,138 @@ error_path=$(unmask_filename "$SBATCH_ERROR")
+					Items: []batchv1.Job{
+						*wrappers.MakeJob("", metav1.NamespaceDefault).
+							Completions(1).
+							CompletionMode(batchv1.IndexedCompletion).
+							Profile("profile").
+							Mode(v1alpha1.SlurmMode).
+							WithContainer(*wrappers.MakeContainer("c1", "bash:4.4").
+								Command("bash", "/slurm/scripts/entrypoint.sh").
+								WithResources(corev1.ResourceRequirements{
+									Limits: corev1.ResourceList{
+										corev1.ResourceMemory: resource.MustParse("1G"),
+									},
+								}).
+								Obj()).
+							WithContainer(*wrappers.MakeContainer("c2", "bash:4.4").
+								Command("bash", "/slurm/scripts/entrypoint.sh").
+								WithResources(corev1.ResourceRequirements{
+									Limits: corev1.ResourceList{
+										corev1.ResourceMemory: resource.MustParse("1G"),
+									},
+								}).
+								Obj()).
+							Obj(),
+					},
+				},
+				&corev1.ConfigMapList{},
+			},
+			cmpopts: []cmp.Option{
+				cmpopts.IgnoreFields(corev1.LocalObjectReference{}, "Name"),
+				cmpopts.IgnoreFields(metav1.OwnerReference{}, "Name"),
+				cmpopts.IgnoreFields(corev1.PodSpec{}, "InitContainers"),
+				cmpopts.IgnoreTypes([]corev1.EnvVar{}),
+				cmpopts.IgnoreTypes([]corev1.Volume{}),
+				cmpopts.IgnoreTypes([]corev1.VolumeMount{}),
+				cmpopts.IgnoreTypes(corev1.ConfigMapList{}),
+			},
+			wantOutPattern: `job\.batch\/.+ created\\nconfigmap\/.+ created`,
+		},
+		"should handle non-exact --mem division across containers": {
+			beforeTest: beforeSlurmTest,
+			afterTest:  afterSlurmTest,
+			args: func(tc *createCmdTestCase) []string {
+				return []string{
+					"slurm",
+					"--profile", "profile",
+					"--",
+					"--mem", "1G",
+					tc.tempFile,
+				}
+			},
+			kjobctlObjs: []runtime.Object{
+				wrappers.MakeJobTemplate("slurm-job-template", metav1.NamespaceDefault).
+					WithContainer(*wrappers.MakeContainer("c1", "bash:4.4").Obj()).
+					WithContainer(*wrappers.MakeContainer("c2", "bash:4.4").Obj()).
+					Obj(),
+				wrappers.MakeApplicationProfile("profile", metav1.NamespaceDefault).
+					WithSupportedMode(*wrappers.MakeSupportedMode(v1alpha1.SlurmMode, "slurm-job-template").Obj()).
+					Obj(),
+			},
+			gvks: []schema.GroupVersionKind{
+				{Group: "batch", Version: "v1", Kind: "Job"},
+				{Group: "", Version: "v1", Kind: "ConfigMap"},
+			},
+			wantLists: []runtime.Object{
+				&batchv1.JobList{
+					TypeMeta: metav1.TypeMeta{Kind: "JobList", APIVersion: "batch/v1"},
+					Items: []batchv1.Job{
+						*wrappers.MakeJob("", metav1.NamespaceDefault).
+							Completions(1).
+							CompletionMode(batchv1.IndexedCompletion).
+							Profile("profile").
+							Mode(v1alpha1.SlurmMode).
+							WithContainer(*wrappers.MakeContainer("c1", "bash:4.4").
+								Command("bash", "/slurm/scripts/entrypoint.sh").
+								WithResources(corev1.ResourceRequirements{
+									Limits: corev1.ResourceList{
+										corev1.ResourceMemory: resource.MustParse("0.5G"),
```

Comment:
Why is this specified differently between containers? Can we align it? If we choose 0.5G test if the partial settings work ok e2e. I would suppose it is safer to use the lower unit as below: 500M

Reply:
Yeah, It seems it returns the lower unit... Changed to M. Thanks

Reply:
Ah, ok so this was just test assert that was inconsistent, thanks for the update.